pkg/devfile/parser/data/2.0.0/components.go

```diff
@@ -227,3 +228,84 @@ func (d *Devfile200) UpdateEvents(postStart, postStop, preStart, preStop []strin
 		d.Events.PreStop = preStop
 	}
 }
+
+// AddVolume adds the volume to the devFile and mounts it to all the container components
+func (d *Devfile200) AddVolume(volume common.Volume, path string) error {
+	for _, component := range d.Components {
+		if component.Container != nil {
+			for _, volumeMount := range component.Container.VolumeMounts {
+				if volumeMount.Path == path {
+					return fmt.Errorf("another volume is mounted on the same path")
+				}
+			}
+			component.Container.VolumeMounts = append(component.Container.VolumeMounts, common.VolumeMount{
+				Name: volume.Name,
+				Path: path,
+			})
+		}
+		if component.Volume != nil && component.Volume.Name == volume.Name {
+			return fmt.Errorf("volume %s already exists", volume.Name)
+		}
+	}
+
+	d.Components = append(d.Components, common.DevfileComponent{
+		Volume: &volume,
+	})
+
+	return nil
+}
+
+// DeleteVolume removes the volume from the devFile and removes all the related volume mounts
+func (d *Devfile200) DeleteVolume(name string) error {
+	found := false
+	for i := len(d.Components) - 1; i >= 0; i-- {
+		if d.Components[i].Container != nil {
+			var tmp []common.VolumeMount
+			for _, volumeMount := range d.Components[i].Container.VolumeMounts {
+				if volumeMount.Name != name {
+					tmp = append(tmp, volumeMount)
+				}
+			}
+			d.Components[i].Container.VolumeMounts = tmp
+		}
+
+		if d.Components[i].Volume != nil {
+			if d.Components[i].Volume.Name == name {
+				found = true
+				d.Components = append(d.Components[:i], d.Components[i+1:]...)
+			}
+		}
+	}
+
+	if !found {
+		return fmt.Errorf("volume %s is not found", name)
+	}
+
+	return nil
+}
+
+// GetVolumeMountPath gets the mount path of the required volume
+func (d *Devfile200) GetVolumeMountPath(name string) (string, error) {
+	volumeFound := false
+	mountFound := false
+	path := ""
+
+	for _, component := range d.Components {
+		if component.Container != nil {
+			for _, volumeMount := range component.Container.VolumeMounts {
+				if volumeMount.Name == name {
+					mountFound = true
+					path = volumeMount.Path
+				}
+			}
+		} else if component.Volume != nil {
+			volumeFound = true
+		}
+	}
+	if volumeFound && mountFound {
```

Comment:
Hope I don't embarass myself with this one, but IIUC, `mountFound` and `volumeFound` cannot be `true` at the same time since they're being set to `true` in opposite situations (one in `if` and another in `else`). Unless we make it:
```go
if component.Volume != nil{
```
instead of:
```go
else if component.Volume != nil
````

Reply:
> but IIUC, mountFound and volumeFound cannot be true at the same time since they're being set to true in opposite situations (one in if and another in else)

They can be true at the same time too. A component can be a volume or a container. Since we are looping over the components, one can be volume (volumeFound=true) and the other can be a container, which has the volume mounted to it (mountFound=true).